src/test/java/uk/gov/hmcts/reform/feature/service/JwtParserServiceTest.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.reform.feature.service;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.junit.Test;
+import uk.gov.hmcts.reform.feature.model.UserTokenDetails;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JwtParserServiceTest {
+
+    private static final String SIGNING_KEY = "signing";
+
+    private final JwtParserService service = new JwtParserService(Jwts.parser().setSigningKey(SIGNING_KEY));
+
+    @Test
+    public void should_parse_valid_jwt_token_and_get_user_details() {
+        // given
+        String jwtToken = Jwts.builder()
+            .signWith(SignatureAlgorithm.HS256, SIGNING_KEY)
+            .addClaims(ImmutableMap.of("id", "1234", "data", "beta,test"))
```

Comment:
as before, idam seems to be putting roles in `roles` claim
https://github.com/hmcts/idam-app/blob/master/api/src/main/java/uk/gov/hmcts/idam/infrastructure/security/authentication/UserAuthenticator.java#L287-L301